### PR TITLE
sql: make COPY TO/FROM respect timeouts and cancellation, and appear in SHOW QUERIES

### PR DESCRIPTION
**Pull request by rafiss**
informs https://github.com/cockroachdb/cockroach/issues/94194

### sql: use conn executor state machine for COPY FROM 

Using the conn executor state machine allows a lot of bespoke logic to
be removed. (If we were to remove the copy_from_atomic_enabled setting
and make it always true, we could remove even more custom logic.)

This then makes it easy to hook up to cancellation and introspection.

Release note (bug fix): The COPY FROM command now respects the
statement_timeout and transaction_timeout settings.

Release note (bug fix): The COPY FROM command now appears in the results
of SHOW QUERIES.

---

### sql: make stmt/txn timeout apply to COPY TO

This commit simplifies the conn executor state transitions a bit, and
adds in timers for stmt/txn timeouts analogously to how they were for
COPY FROM.

No release note since COPY TO is new.

### CODE REVIEW

**Comment by otan** [comment]
cc @cucaroach in case this influences any work you're doing on copy statement bundles

**Review by otan — COMMENTED** [review]
did a first pass: some tests look like they need some fixing.

i also have a minor concern about how this affects retries - COPY FROM / TO should never retry (given it directly digests / spits out messages on the wire without buffering them). do you know if that is still valid / a test case already covers them?

**Comment by rafiss** [comment]
yes, there are tests. see `TestCopyFromRetries`

File: pkg/sql/copy_from.go [hunk 1]
```diff
@@ -792,37 +792,23 @@ func (c *copyMachine) readBinarySignature() ([]byte, error) {
...
 //
-// It returns a cleanup function that needs to be called when we're
-// done with the planner (before preparePlannerForCopy is called
-// again). The cleanup function commits the txn (if it hasn't already
-// been committed) or rolls it back depending on whether it is passed
-// an error. If an error is passed in to the cleanup function, the
-// same error is returned.
+// It returns a cleanup function that needs to be called when we're done with
+// the planner (before preparePlannerForCopy is called again). If
+// CopyFromAtomicEnabled is false, The cleanup function commits the txn (if it
```
Comment (cucaroach) [inline pkg/sql/copy_from.go:797]: s/The/the/

**Comment by rafiss** [comment]
i've fixed those test failures; the changes were pretty minor

**Review by otan — APPROVED** [review]

**Comment by rafiss** [comment]
tftr!

bors r=otan

**Pull request merged**